test/jdk/java/foreign/TestLinker.java

```diff
@@ -151,6 +155,16 @@ public void testCanonicalLayouts(String typeName) {
         assertTrue(layout instanceof ValueLayout);
     }
 
+    @Test
+    public void embeddedPaddingLayout() {
+        PaddingLayout padding = MemoryLayout.paddingLayout(64).withByteAlignment(64);
+        SequenceLayout sequence = MemoryLayout.sequenceLayout(2, padding);
+        StructLayout struct = MemoryLayout.structLayout(sequence);
+        FunctionDescriptor fd = FunctionDescriptor.of(struct, struct);
+        Linker linker = Linker.nativeLinker();
+        assertThrows(IllegalArgumentException.class, () -> linker.downcallHandle(fd));
```

Comment:
Could you please verify the exception message here as well? (Since there are multiple IAEs possible)